### PR TITLE
Disable thrift by default

### LINKED ISSUE #8336: Do not enable the Thrift server by default
Currently, the Thrift port (default: 9160) is always open when Scylla starts, but it's not clear if anyone actually uses it. Since this layer, while functional, is not tested and maintained as actively as CQL, it should be off by default.

### PR DESCRIPTION
**Pull request by psarna**
The Thrift layer is functional, but it's not usually the first-choice protocol for Scylla users, so it's hereby disabled by default.

Fixes #8336

### CODE REVIEW

**Comment by denesb** [comment]
Make sure to follow-up with updates to thrift related dtests.

**Comment by nyh** [comment]
> Make sure to follow-up with updates to thrift related dtests.

Good point.
This "follow up" probably needs to be done **before** committing this patch, otherwise those tests will after this patch... (to be honest, I don't even know if we have thrift dtests, I just assume we do... @psarna please check).

File: docs/design-notes/protocols.md [hunk 1]
```diff
@@ -166,6 +166,10 @@ but was recently dropped in Cassandra (version 4.0) and is likely to be
 dropped by Scylla in the future as well, so it is not recommended for new
 applications.
 
+By default, Scylla does not enable the Thrift server. In order to use it,
+it must be explicitly enabled by setting the `start_rpc` configuration option
+to true.
+
 By default scylla listens to the Thrift protocol on port 9160, which can be
```
Comment (nyh) [inline docs/design-notes/protocols.md:173]: Maybe add a "When Thirft is enabled," in the beginning of this sentence, to clarify it doesn't contradict the new paragraph you added right before it.

**Review by nyh — APPROVED** [review]

**Comment by denesb** [comment]
> > Make sure to follow-up with updates to thrift related dtests.
> 
> Good point.
> This "follow up" probably needs to be done **before** committing this patch, otherwise those tests will after this patch... (to be honest, I don't even know if we have thrift dtests, I just assume we do... @psarna please check).

We do, I broke some in the past. :)

**Comment by psarna** [comment]
@denesb @nyh I believe that Thrift-related dtests already conveniently enabled this layer explicitly (see: https://github.com/scylladb/scylla-dtest/commit/a69743d86f9c7045fc5334389b2c1f5307867451) . I also grepped for gating dtests that also use Thrift and I only found one in `thrift_tests.py`, which has Thrift enabled.

**Comment by psarna** [comment]
v2:
 - added a "When Thrift is enabled" intro in the next paragraph in docs

**Comment by psarna** [comment]
ping

**Pull request merged**